common/txmgr/types/tx_store.go

```diff
@@ -81,6 +82,10 @@ type TxStore[
 	Close()
 }
 
+type ReaperTxStore[CHAIN_ID ID] interface {
+	ReapConfirmedOrFatalErrorTxs(minBlockNumberToKeep int64, timeThreshold time.Time, chainID CHAIN_ID) error
```

Comment:
If feels backwards for this to have `TxStore` in the name. If we could come up with a short term for `ConfirmedOrFatalError(Txs)`, then it could be named more idiomatically:
```suggestion
type XYZReaper[CHAIN_ID ID] interface {
	ReapXYZTxs(minBlockNumberToKeep int64, timeThreshold time.Time, chainID CHAIN_ID) error
```

Reply:
Or would it make sense just as the _reaper_ store?
```suggestion
type ReaperStore[CHAIN_ID ID] interface {
	ReapConfirmedOrFatalErrorTxs(minBlockNumberToKeep int64, timeThreshold time.Time, chainID CHAIN_ID) error
```

Reply:
```suggestion
type TxHistoryReaper[CHAIN_ID ID] interface {
	ReapTxHistory(minBlockNumberToKeep int64, timeThreshold time.Time, chainID CHAIN_ID) error
```

Reply:
I like this one.